cosmovisor/cmd/cosmovisor/cmd/help.go

```diff
@@ -0,0 +1,101 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/cosmovisor"
+	"os"
+)
+
+// ShouldGiveHelp checks the env and provided args to see if help is needed or being requested.
+// Help is needed if the os.Getenv(EnvName) env var isn't set.
+// Help is requested if the first arg is "help" or "--help"; or the only arg is "-h"
+func ShouldGiveHelp(args []string) bool {
+	if len(os.Getenv(cosmovisor.EnvName)) == 0 {
+		return true
+	}
+	if len(args) == 0 {
+		return false
+	}
+	return args[0] == "help" || args[0] == "--help" || (len(args) == 1 && args[0] == "-h")
+}
+
+// DoHelp outputs help text, config info, and attempts to run the binary with the --help flag.
+func DoHelp() {
+	// Output the help text
+	fmt.Println(GetHelpText())
+	// If the config isn't valid, say what's wrong and we're done.
+	cfg, cerr := cosmovisor.GetConfigFromEnv()
+	switch err := cerr.(type) {
+	case nil:
+		// Nothing to do. Move on.
+	case *cosmovisor.MultiError:
+		fmt.Fprintf(os.Stderr, "[cosmovisor] multiple configuration errors found:\n")
```

Comment:
@spoo-bar  is merging a logger hopefully tomorrow. Let's update this PR after the merge and use a logger instead of `fmt.Fprint`.

Reply:
Since this is part of the help flow, I'm not sure if it's something that needs to be logged, and it's definitely something that should go to either `stderr` or `stdout`.

But I'll take a look once the logger gets merged in.

Reply:
@spoo-bar  merged the logger: https://github.com/cosmos/cosmos-sdk/pull/10217